cpp/include/cudf/stream_compaction.hpp

```diff
@@ -329,6 +330,7 @@ std::unique_ptr<table> stable_distinct(
   duplicate_keep_option keep        = duplicate_keep_option::KEEP_ANY,
   null_equality nulls_equal         = null_equality::EQUAL,
   nan_equality nans_equal           = nan_equality::ALL_EQUAL,
+  rmm::cuda_stream_view stream      = cudf::get_default_stream(),
```

Comment:
Okay so now the public API has exactly the same parameters as the detail API. Should we remove the detail API now?

Reply:
The public API should have an nvtx call and the detail would not.
Same answer as when you asked this before
https://github.com/rapidsai/cudf/pull/14263#discussion_r1351459116

Reply:
Sorry my bad. Didn't remember that.

Reply:
Are they really the same? I'd expect the detail one to not have a default value for the stream parameter.

Reply:
That is true. The detail APIs should not have any default parameters.
That one is documented: https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#default-parameters

Reply:
Same signature (set of parameters) but the detail one does not have default values.